mkosi/__init__.py

```diff
@@ -2218,21 +2218,61 @@ def finalize_initrds(context: Context) -> list[Path]:
     return [build_default_initrd(context)]
 
 
+def expand_kernel_specifiers(text: str, kver: str, token: str, roothash: str, boot_count: str) -> str:
+    specifiers = {
+        "&": "&",
+        "e": token,
+        "k": kver,
+        "h": roothash,
+        "c": boot_count
+    }
+
+    def replacer(match: re.Match[str]) -> str:
+        m = match.group("specifier")
+        if specifier := specifiers.get(m):
+            return specifier
+
+        logging.warning(f"Unknown specifier '&{m}' found in {text}, ignoring")
+        return ""
+
+    return re.sub(r"&(?P<specifier>[&a-zA-Z])", replacer, text)
+
+
 def install_type1(
     context: Context,
     kver: str,
     kimg: Path,
     token: str,
     partitions: Sequence[Partition],
 ) -> None:
-    dst = context.root / "boot" / token / kver
-    entry = context.root / f"boot/loader/entries/{token}-{kver}.conf"
+    roothash = ""
+    if roothash_cli := finalize_roothash(partitions):
+        roothash = roothash_cli.partition("=")[2]
+
+    boot_count = ""
+    if (context.root / "etc/kernel/tries").exists():
+        boot_count = (context.root / "etc/kernel/tries").read_text().strip()
+
+    bootloader_entry = context.config.bootloader_entry_format or "&e-&k"
+    bootloader_entry_dir = bootloader_entry.replace("-", "/")
```

Comment:
I also wasn't sure really if this made sense, but since @DaanDeMeyer did also want it for Type1 I at least tried to do something that made a bit of sense (although I also wasn't happy about the substitution)

Reply:
Ah, I missed that, thanks!

I don't think we should do the replacement arbitrarily deeply, because then anybody not using the default for Type 1 entries would end up with an arbitrarily deep directory structure. I see multiple alternatives
- Only replace the first separator (`-` currently, maybe `_` in the future), i.e. `.replace("-", "/", count=1)
- Only replace the separator after `&e`, i.e. `.replace("&e-", "&e/", count=1)`
- Force an (additional?) `&e/` before the name.

Reply:
I think option 1 makes the most sense.

Reply:
I've been thinking about this again. 

Would there be any problem with tying this to `systemd-escape --unescape {bootloader_entry_format}`? This would probably be "the most familiar" way to handle such a path  conversion.